retrieval/target.go

```diff
@@ -359,9 +364,15 @@ func (t *Target) scrape(sampleAppender storage.SampleAppender) (err error) {
 	}()
 
 	for samples := range t.ingestedSamples {
-		for _, s := range samples {
-			s.Metric.MergeFromLabelSet(baseLabels, clientmodel.ExporterLabelPrefix)
-			sampleAppender.Append(s)
+		if t.proxy {
+			for _, s := range samples {
+				sampleAppender.Append(s)
```

Comment:
Yeah, so have three different `mode`s:

1. normal, as we have it now
2. honor labels, i.e. don't resolve clashes with auto-attached labels by `exporter_xxx` but leave the original label intact.
3. don't auto-attach any labels

We just need real good (tm) names for those...